File: app/src/main/java/io/mgba/UI/Fragments/Interfaces/IGamesFragment.java
package io.mgba.UI.Fragments.Interfaces;

import java.util.List;

import io.mgba.Data.Database.Game;

public interface IGamesFragment<T> {
    void swapContent(List<T> items);
    void stopRefreshing();
    void showContent(boolean state);
    void handleItemClick(Game game);
}

File: app/src/main/java/io/mgba/ui/fragments/interfaces/IGamesFragment.kt
package io.mgba.ui.fragments.interfaces

import io.mgba.data.local.database.model.Game

interface IGamesFragment<T> {
    fun swapContent(items: List<T>)
    fun stopRefreshing()
    fun showContent(state: Boolean)
    fun handleItemClick(game: Game)
}
